services/proxy/pkg/middleware/policies_test.go

```diff
@@ -0,0 +1,229 @@
+		g.Expect(r.Environment.Request.Method).To(Equal(http.MethodDelete))
+		g.Expect(r.Environment.Request.Path).To(Equal("/whatever"))
+	})
+}
+
+func TestPolicies_EvaluationEnvironment_Resource(t *testing.T) {
+	pmt := newPoliciesMiddlewareTester(t)
+
+	// tus metadata
+	pmt.httpRequest = httptest.NewRequest(http.MethodPost, "/remote.php/dav/spaces", nil)
+	pmt.httpRequest.Header.Set(net.HeaderUploadMetadata, fmt.Sprintf("filename %v", base64.StdEncoding.EncodeToString([]byte("tus-file-name.png"))))
+
+	pmt.test(func(g *WithT, _ *httptest.ResponseRecorder, r *policiesPG.EvaluateRequest, _ *policiesPG.EvaluateResponse) {
+		g.Expect(r.Environment.Resource.Name).To(Equal("tus-file-name.png"))
+	})
+
+	// url path
+	pmt.httpRequest = httptest.NewRequest(http.MethodPut, "/remote.php/dav/spaces/simple-file-name.png", nil)
+	pmt.test(func(g *WithT, _ *httptest.ResponseRecorder, r *policiesPG.EvaluateRequest, _ *policiesPG.EvaluateResponse) {
+		g.Expect(r.Environment.Resource.Name).To(Equal("simple-file-name.png"))
+	})
+
+	// shared-resource put
+	pmt.httpRequest = httptest.NewRequest(http.MethodPut, "/remote.php/dav/spaces/897987fd978dffdfds9f78dsf97fd", nil)
+	pmt.gwClientStatResponse.Info.Name = "shared-file-name.png"
+	pmt.test(func(g *WithT, _ *httptest.ResponseRecorder, r *policiesPG.EvaluateRequest, _ *policiesPG.EvaluateResponse) {
+		g.Expect(r.Environment.Resource.Name).To(Equal("shared-file-name.png"))
+	})
+}
+
+func TestPolicies_NoQuery_PassThrough(t *testing.T) {
+	pmt := newPoliciesMiddlewareTester(t)
+	pmt.regoQuery = ""
+
+	pmt.test(func(g *WithT, w *httptest.ResponseRecorder, _ *policiesPG.EvaluateRequest, _ *policiesPG.EvaluateResponse) {
+		g.Expect(w.Code).To(Equal(http.StatusOK))
+	})
+}
+
+func newPoliciesMiddlewareTester(t *testing.T) policiesMiddlewareTester {
+	return policiesMiddlewareTester{
+		g:           NewWithT(t),
+		regoQuery:   "any",
+		httpRequest: httptest.NewRequest(http.MethodGet, "/policies", nil),
+		eval: func(request *policiesPG.EvaluateRequest, response *policiesPG.EvaluateResponse) error {
+			return nil
+		},
+		gwClientStatResponse: &sprovider.StatResponse{
+			Status: status.NewOK(context.Background()),
+			Info:   &sprovider.ResourceInfo{},
+		},
+	}
+}
+
+type policiesMiddlewareTester struct {
+	g                    *WithT
+	regoQuery            string
+	httpRequest          *http.Request
+	grpcEvaluateRequest  *policiesPG.EvaluateRequest
+	grpcEvaluateResponse *policiesPG.EvaluateResponse
+	gwClientStatResponse *sprovider.StatResponse
+	eval                 func(request *policiesPG.EvaluateRequest, response *policiesPG.EvaluateResponse) error
+}
+
+func (pmt *policiesMiddlewareTester) test(e func(g *WithT, w *httptest.ResponseRecorder, grpcRequest *policiesPG.EvaluateRequest, grpcResponse *policiesPG.EvaluateResponse)) {
+	var (
+		polServiceAddress   = fmt.Sprintf("127.0.0.1:%d", freeport.GetPort()) //non-ephemeral port didn't work
```

Comment:
Mmh. Do we really need to find a free port to boot this up? Can't we just mock policies service? (Like we do with gateway?)

Reply:
nope we don't if we use the mock client instead, the idea was to not use it and use a real grpc server instead, not a fan?

Reply:
fixed, i use mocks instead.